mongodump/mongodump_test.go

```diff
@@ -579,9 +587,47 @@ func TestMongoDumpBSON(t *testing.T) {
 					md.OutputOptions.Archive = "dump_slash.archive"
 					err = md.Init()
 					So(err, ShouldBeNil)
+				})
+			})
+
+			Convey("and that it dumps a collection with a name >238 bytes in the right format", func() {
+				session, err := testutil.GetBareSession()
+				So(err, ShouldBeNil)
+				coll := session.Database(testDB).Collection(longCollectionName)
+				_, err = coll.InsertOne(nil, bson.M{"a": 1})
+				So(err, ShouldBeNil)
+				defer coll.Drop(nil)
+
+				md.ToolOptions.Namespace.Collection = longCollectionName
+				err = md.Init()
+				So(err, ShouldBeNil)
 
+				path, err := os.Getwd()
+				So(err, ShouldBeNil)
+
+				absDumpDir := util.ToUniversalPath(filepath.Join(path, "dump_slash"))
+				So(os.RemoveAll(absDumpDir), ShouldBeNil)
+				So(fileDirExists(absDumpDir), ShouldBeFalse)
+
+				dumpDBDir := util.ToUniversalPath(filepath.Join("dump_slash", testDB))
+				So(fileDirExists(dumpDBDir), ShouldBeFalse)
+
+				md.OutputOptions.Out = "dump_slash"
+				err = md.Dump()
+				So(err, ShouldBeNil)
+				So(fileDirExists(dumpDBDir), ShouldBeTrue)
+
+				Convey("to a bson file", func() {
+					oneBsonFile, err := os.Open(util.ToUniversalPath(filepath.Join(dumpDBDir, longBsonName)))
+					So(err, ShouldBeNil)
+					defer oneBsonFile.Close()
```

Comment:
Deferring this right at the end of the function seems a bit pointless. I think you can just close it.